sound/soc/intel/boards/sof_board_helpers.c

```diff
@@ -624,6 +778,9 @@ sof_intel_board_get_ctx(struct device *dev, unsigned long board_quirk)
 				SOF_SSP_PORT_BT_OFFLOAD_SHIFT;
 	}
 
+	if (board_quirk & SOF_HDA_CODEC_PRESENT)
+		ctx->hda_codec_present = true;
```

Comment:
Should we set `ctx->hda_codec_present` by HDA_EXT_CODEC(codec_mask)?

Reply:
It's done in machine driver since we didn't pass codec_mask to board helper module.

Reply:
> It's done in machine driver since we didn't pass codec_mask to board helper module.

Can we set `ctx->hda_codec_present` in machine driver? Just like [ctx->hdmi.idisp_codec = true;](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/skl_hda_dsp_generic.c#L232) I am thinking is SOF_HDA_CODEC_PRESENT really needed?

Reply:
sure, we could remove that flag.